File: backend/le1010274/Schedule/GetSchedule.py
import json
import sys


def main():

    fileName = sys.argv[1]
    #fileName = 'tuesday.json'
    state = sys.argv[2]
    #state = 'Illinois'
    city = sys.argv[3]
    #city = 'Chicago'
    store = sys.argv[4]
    #store = 'United:Supermarket'
    address = sys.argv[5]
    #address = '1234:This:Way'

    state = state.replace(":", "-")
    city = city.replace(":", "-")
    store = store.replace(":", "-")
    address = address.replace(":", "-")
    #storeusername = storeusername.replace(":", "-")
    # print(state)
    filePath = "/var/www/html/cs4391/le1010274/Schedule/"
    filePath += state + "/" + city + "/" + store + "/" + address + "/"
    # filePath += storeUsername + "/" + fileName
    filePath += fileName


    with open(filePath) as f:
        data = json.load(f)
    f.close()

    schedule = ""
    temp = list(data.keys())
    temp.sort()
    for key in temp:
        if key == 'capacity':
            schedule += key
            schedule += ':'
            schedule += str(data[key])
            schedule += ','
        # end if
        else:
            schedule += key
            schedule += ":"
            schedule += str(data[key]["num_reservations"])
            schedule += ","
        #print(schedule)
    # end for


    print(schedule)

if __name__ == "__main__":
    main()

File: backend/le1010274/Schedule/GetSchedule.php
<?php
   header("Access-Control-Allow-Origin: {$_SERVER['HTTP_ORIGIN']}");
   header("Access-Control-Allow-Headers: {$_SERVER['HTTP_ACCESS_CONTROL_REQUEST_HEADERS']}");

   $fileName = $_POST['fileName'];
   //$fileName = 'monday.json';
   $state = $_POST['state'];
   //$state = 'Texas';
   $city = $_POST['city'];
   //$city = 'Amarillo';
   $store = $_POST['store'];
   //$store = 'Rando Mart';
   $address = $_POST['address'];
   //$address = '3 Lancaster Road';

   $state = str_replace(" ", ":", $state);
   $city = str_replace(" ", ":", $city);
   $store = str_replace(" ", ":", $store);
   $address = str_replace(" ", ":", $address);
    
   $command = escapeshellcmd('python /var/www/html/cs4391/le1010274/Schedule/GetSchedule.py');
   $command = $command." ".$fileName;
   $command = $command." ".$state;
   $command = $command." ".$city;
   $command = $command." ".$store;
   $command = $command." ".$address;

   //echo $command;


   $output = shell_exec($command);
   echo $output;


?>

